units/host_monitoring_parent_decommission.go

```diff
@@ -0,0 +1,134 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/distro"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/pkg/errors"
+)
+
+const (
+	parentDecommissionJobName = "parent-decommission-job"
+)
+
+func init() {
+	registry.AddJobType(parentDecommissionJobName, func() amboy.Job {
+		return makeParentDecommissionJob()
+	})
+
+}
+
+type parentDecommissionJob struct {
+	job.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
+	distroId string
+}
+
+func makeParentDecommissionJob() *parentDecommissionJob {
+	j := &parentDecommissionJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    parentDecommissionJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+func NewParentDecommissionJob(id string, distroId string) amboy.Job {
```

Comment:
I might just pass a *distro.Distro, cache the object in the job in a private attribute, and then save the distroID in an exported field.